lib/storage/metadata/mongoclient/ListRecordStream.js

```diff
@@ -108,9 +108,17 @@ class ListRecordStream extends stream.Readable {
             if (value && value.tags) {
                 value.tags = unescape(value.tags);
             }
+            // updates overwrite the whole metadata,
+            // so they are considered as puts
+            let type = 'put';
+            // we perform an update operation before each
+            // delete to keep the metadata in the oplog
+            // this operation should be considered a delete
+            if (value && value.deleted) {
+                type = 'delete';
+            }
             entry = {
-                type: 'put', // updates overwrite the whole metadata,
-                // so they are considered as puts
+                type,
```

Comment:
should we not also filter the "delete" event accordingly (e.g. line 131) ?

Reply:
I'm afraid ignoring all deletion events would mean loosing delete events for all other types of collections (__metastore, ...) if we ever decide to use that. The update->delete logic is only done on bucket "type" collections.

Reply:
would be nice to add a comment for posterity, to explain why we cannot filter the "real" delete event and why this is not a problem (e.g. it will be filtered by LogReader in Backbeat, as the event will not have any metadata)